db/migrate/20210212223928_handle_nil_variants.rb

```diff
@@ -0,0 +1,56 @@
+class HandleNilVariants < ActiveRecord::Migration
+   def up
+     Spree::Variant.where(product_id: nil).destroy_all
+     Spree::Variant.where(unit_value: [nil, Float::NAN]).find_each do |variant|
```

Comment:
I just checked this on UK prod:
```
irb(main):004:0> Spree::Variant.where(weight: [nil, Float::NAN]).count
=> 74008
```
74 thousand..?

Reply:
```
irb(main):005:0> Spree::Variant.where(weight: Float::NAN).count
=> 0
```

Reply:
I think this work has morphed quite a bit over the course of this PR (and the previous attempts); looking at it now, the most important thing, I'd say, is to get the `<> NaN` constraints on the database table. I think leaving variants with a `nil` weight is okay. I think `unit_value` should always be present and default to 1, but weight can be nil. 

I changed the migration to allow this in 05d49f76d51fd69a9ae7337bae240ca829a6736c; it's now running in under two minutes on my machine.